helmfile.d/charts/prometheus-alerts/files/cluster-autoscaler.yaml

```diff
@@ -0,0 +1,142 @@
+      for: 1m
+      labels:
+        severity: high
+      annotations:
+        summary: "Cluster Autoscaler failed to scale up GPU nodes"
+        description: "The Cluster Autoscaler for cluster {{"`{{ $labels.cluster }}`"}} encountered an error while attempting to scale up GPU nodes. Reason: {{"{{ $labels.reason }}"}}, GPU Resource: {{"{{ $labels.gpu_resource_name }}"}}, GPU Name: {{"{{ $labels.gpu_name }}"}}"
+        runbook_url: {{ .Values.runbookUrls.clusterAutoscaler.ClusterAutoscalerFailedGPUScaleUp }}
+
+    - alert: ClusterAutoscalerNodeGroupUnhealthy
+      expr: cluster_autoscaler_node_group_healthiness == 0
+      for: 10m
+      labels:
+        severity: warning
+      annotations:
+        summary: "Unhealthy node group"
+        description: "The node group {{"{{ $labels.node_group }}"}} in cluster {{"`{{ $labels.cluster }}`"}} is reporting as unhealthy which might prevent autoscaling operations for this group."
+        runbook_url: {{ .Values.runbookUrls.clusterAutoscaler.ClusterAutoscalerNodeGroupUnhealthy }}
+
+    - alert: ClusterAutoscalerNodeGroupBackoff
+      expr: cluster_autoscaler_node_group_backoff_status == 1
+      for: 5m
+      labels:
+        severity: warning
+      annotations:
+        summary: "A node group is in backoff state"
+        description: "The node group {{"{{ $labels.node_group }}"}} in cluster {{"`{{ $labels.cluster }}`"}} is currently in a backoff state due to reason: {{"{{ $labels.reason }}"}}. This means that the Cluster Autoscaler will not try to scale it up."
+        runbook_url: {{ .Values.runbookUrls.clusterAutoscaler.ClusterAutoscalerNodeGroupBackoff }}
+
+    - alert: ClusterAutoscalerUnneededNodesNotScaledDown
+      expr: cluster_autoscaler_unneeded_nodes_count > 0
+      for: 24h
+      labels:
+        severity: warning
+      annotations:
+        summary: "Cluster Autoscaler reports unneeded nodes that are not being scaled down"
+        description: "There are {{"{{ $value }}"}} nodes in cluster {{"`{{ $labels.cluster }}`"}} identified as unneeded by the Cluster Autoscaler, but they have not been scaled down for 24 hours."
+
+    # TODO: The reason label is just the reason constant integer, needs to be fixed upstream. Perhaps a new label like "reason_str" or propose to change it to the name of the constant.
+    # TODO: Doesn't tell you which node it is. Needs to be fixed upstream.
+    # - alert: ClusterAutoscalerUnremovableNodesStuck
+    #   expr: cluster_autoscaler_unremovable_nodes_count > 0
+    #   for: 15m
+    #   labels:
+    #     severity: warning
+    #   annotations:
+    #     summary: "Cluster Autoscaler reports unremovable nodes"
+    #     description: "There are {{"{{ $value }}"}} node(s) that the Cluster Autoscaler cannot remove due to reason: {{"{{ $labels.reason }}"}}."
+    #     runbook_url: {{ .Values.runbookUrls.clusterAutoscaler.ClusterAutoscalerUnremovableNodesStuck }}
```

Comment:
I'm not sure what to do about this one. I think it can be really useful if tweaked correctly but right the alerts aren't very informative (see the TODO comments). Do you want to keep it here for the future or should I get rid of it?

It should correspond to the `Unremovable nodes` pane in the Grafan dashboard.

Reply:
I still think that it's usable. Then you at least know that you need to look into it. Because I assume that the reason will show up in like the autoscaler logs or so.
However, I do agree that it would be more usable if we had some better reason, and more importantly, what node directly in the alert

Reply:
> Riiight 🤦 And it's not in any status of the nodes or something?

Not that I'm aware of. I think this lives entirely in the Cluster Autoscaler's internal state.

Reply:
But should I leave it as a TODO or remove it entirely?

Reply:
I think that leaving it as a TODO is good. Because this feels like we want to have if they ever add the reason as a string